include/aspect/postprocess/tracer.h

```diff
@@ -35,36 +38,125 @@ namespace aspect
+         */
+        ~PassiveTracers();
+
+        /**
+         *
+         */
+        void
+        initialize();
+
+        /**
+         * Returns a reference to the particle world, in case anyone wants to
+         * query something about tracers.
+         */
+        const Particle::World<dim> &
+        get_particle_world() const;
+
+        /**
+         * Returns a reference to the particle world, in case anyone wants to
+         * query something about tracers.
+         */
+        Particle::World<dim> &
+        get_particle_world();
+
+        /**
+         * Execute this postprocessor. Derived classes will implement this
+         * function to do whatever they want to do to evaluate the solution at
+         * the current time step.
+         *
+         * @param[in,out] statistics An object that contains statistics that
+         * are collected throughout the simulation and that will be written to
+         * an output file at the end of each time step. Postprocessors may
+         * deposit data in these tables for later visualization or further
+         * processing.
+         *
+         * @return A pair of strings that will be printed to the screen after
+         * running the postprocessor in two columns; typically the first
+         * column contains a description of what the data is and the second
+         * contains a numerical value of this data. If there is nothing to
+         * print, simply return two empty strings.
+         */
+        virtual
+        std::pair<std::string,std::string> execute (TableHandler &statistics);
+
+        /**
+         * Save the state of this object.
+         */
+        virtual
+        void save (std::map<std::string, std::string> &status_strings) const;
+
+        /**
+         * Restore the state of the object.
+         */
+        virtual
+        void load (const std::map<std::string, std::string> &status_strings);
+
+        /**
+         * Serialize the contents of this class as far as they are not read
+         * from input parameter files.
+         */
+        template <class Archive>
+        void serialize (Archive &ar, const unsigned int version);
+
+        /**
+         * Declare the parameters this class takes through input files.
+         */
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * Read the parameters this class declares from the parameter file.
+         */
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
       private:
         /**
          * The world holding the particles
          */
-        Particle::World<dim, Particle::BaseParticle<dim> >              world;
+        Particle::World<dim>              world;
 
         /**
          * The integrator to use in moving the particles
          */
-        Particle::Integrator::Interface<dim, Particle::BaseParticle<dim> >  *integrator;
+        Particle::Integrator::Interface<dim>  *integrator;
```

Comment:
Where do you get this (and the next few variables) from? I'd be more comfortable if they had a smart pointer type. If you create these variables yourself, use `std_cxx11::unique_ptr`, otherwise `std_cxx11::shared_ptr`.